Adjusting redis docs (#161)
AdobeDocs/commerce-operations.en: help/implementation-playbook/best-practices/planning/redis-service-configuration.md

            compression_lib: 'gzip'       # snappy and lzf for performance, gzip for high compression (~69%)
```

## Enable Redis asynchronous freeing (lazyfree)
To enable `lazyfree` on Adobe Commerce on cloud infrastructure, submit an [Adobe Commerce Support ticket](https://experienceleague.adobe.com/docs/commerce-knowledge-base/kb/help-center-guide/magento-help-center-user-guide.html#submit-ticket) requesting the following Redis configuration be applied to your environment(s):
```
lazyfree-lazy-eviction yes
lazyfree-lazy-expire yes
lazyfree-lazy-server-del yes
replica-lazy-flush yes
lazyfree-lazy-user-del yes
```
When lazyfree is enabled, Redis offloads memory reclamation to background threads for evictions, expirations, server-initiated deletes, user deletes, and replica dataset flushes. This reduces main-thread blocking and can lower request latency.
>[!NOTE]
>

> >`lazyfree-lazy-user-del yes` makes the `DEL` command behave like `UNLINK`, i.e., it unlinks keys immediately and frees their memory asynchronously.

Reviewer: ```suggestion
>The `lazyfree-lazy-user-del yes` option makes the `DEL` command behave like `UNLINK`, which unlinks keys immediately and frees their memory asynchronously.
```

```
io-threads-do-reads yes
io-threads 8 # choose a value lower than the number of CPU cores (check with nproc), then tune under load
```

>[!NOTE]
>
>I/O threads parallelize client I/O and parsing only. Redis command execution remains single-threaded.

>[!WARNING]
>
>Enabling I/O threads can increase CPU usage and does not benefit every workload. Start with a conservative value and benchmark. If latency rises or CPU saturates, reduce `io-threads` or disable reads in I/O threads.

## Increase Redis client timeouts and retries
Raise the cache client’s tolerance to transient saturation by adjusting the backend options in `.magento.env.yaml`:
```
stage:
  deploy:
    CACHE_CONFIGURATION:
      _merge: true
      frontend:
        default:
          backend_options:
            read_timeout: 10
            connect_retries: 5
````

> These settings increase client tolerance to brief congestion on Redis by extending the reply wait window and retrying connection setup. This can reduce intermittent “cannot connect to localhost:6370” and read-timeout errors during short spikes.

Reviewer: ```suggestion
These settings increase client tolerance to brief congestion on Redis by extending the reply wait window and retrying connection setup. This can reduce intermittent `cannot connect to localhost:6370` and read-timeout errors during short spikes.
```
